### ISSUE TITLE
RDS cluster passwords not updating after changes

### ISSUE THREAD

**Issue by caboog** [pulumi/pulumi-aws#2192]
### What happened?

Updating an RDS cluster password via a change in pulumi does not actually take effect. It runs with out error and shows a chance is being made, but the user is still on the old password after running `pulumi up`.

### Steps to reproduce

Create an RDS cluster with instances and set the password using randomPassword and set `applyImmediately: true`
Change the length value with randomPassword to force an updated value.
Apply the change.
Log in to the cluster with the user credentials. `psql -h RDS_ENDPOINT -U DBUSER --dbname postgres -W`

Code to reproduce the issue: https://github.com/caboog/pulumi-examples/tree/main/typescript/aws/ersan-rds-passwords

### Expected Behavior

New password works after applying the change.

### Actual Behavior

Old password is still in use after the change. New password fails to connect.

### Output of `pulumi about`

```
Version      3.37.2
Go Version   go1.17.12
Go Compiler  gc

Plugins
NAME    VERSION
aws     5.18.0
docker  3.4.1
nodejs  unknown
random  4.8.2

Host
OS       darwin
Version  12.6
Arch     arm64

This project is written in nodejs: executable='/opt/homebrew/bin/node' version='v18.11.0'

Current Stack: dev

TYPE                                                 URN
pulumi:pulumi:Stack                                  urn:pulumi:dev::ersan-rds-passwords::pulumi:pulumi:Stack::ersan-rds-passwords-dev
awsx:x:ec2:Vpc                                       urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc::carl
pulumi:providers:aws                                 urn:pulumi:dev::ersan-rds-passwords::pulumi:providers:aws::default_5_18_0
aws:secretsmanager/secret:Secret                     urn:pulumi:dev::ersan-rds-passwords::aws:secretsmanager/secret:Secret::dbpassword11
pulumi:providers:random                              urn:pulumi:dev::ersan-rds-passwords::pulumi:providers:random::default_4_8_2
random:index/randomPassword:RandomPassword           urn:pulumi:dev::ersan-rds-passwords::random:index/randomPassword:RandomPassword::password
awsx:x:ec2:NatGateway                                urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:NatGateway::carl-1
awsx:x:ec2:Subnet                                    urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet::carl-private-1
awsx:x:ec2:Subnet                                    urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet::carl-public-1
awsx:x:ec2:NatGateway                                urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:NatGateway::carl-0
awsx:x:ec2:Subnet                                    urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet::carl-private-0
awsx:x:ec2:InternetGateway                           urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:InternetGateway::carl
aws:ec2/vpc:Vpc                                      urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$aws:ec2/vpc:Vpc::carl
awsx:x:ec2:Subnet                                    urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet::carl-public-0
aws:ec2/eip:Eip                                      urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:NatGateway$aws:ec2/eip:Eip::carl-1
aws:ec2/eip:Eip                                      urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:NatGateway$aws:ec2/eip:Eip::carl-0
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTable:RouteTable::carl-public-0
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTable:RouteTable::carl-private-0
aws:ec2/internetGateway:InternetGateway              urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:InternetGateway$aws:ec2/internetGateway:InternetGateway::carl
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTable:RouteTable::carl-private-1
aws:secretsmanager/secretVersion:SecretVersion       urn:pulumi:dev::ersan-rds-passwords::aws:secretsmanager/secretVersion:SecretVersion::dbpassword-ssm
aws:ec2/subnet:Subnet                                urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/subnet:Subnet::carl-private-1
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTable:RouteTable::carl-public-1
aws:ec2/subnet:Subnet                                urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/subnet:Subnet::carl-public-1
aws:ec2/securityGroup:SecurityGroup                  urn:pulumi:dev::ersan-rds-passwords::aws:ec2/securityGroup:SecurityGroup::carl-test
aws:ec2/subnet:Subnet                                urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/subnet:Subnet::carl-public-0
aws:ec2/subnet:Subnet                                urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/subnet:Subnet::carl-private-0
aws:ec2/route:Route                                  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/route:Route::carl-public-0-ig
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTableAssociation:RouteTableAssociation::carl-private-1
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTableAssociation:RouteTableAssociation::carl-public-1
aws:ec2/route:Route                                  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/route:Route::carl-public-1-ig
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTableAssociation:RouteTableAssociation::carl-public-0
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/routeTableAssociation:RouteTableAssociation::carl-private-0
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:NatGateway$aws:ec2/natGateway:NatGateway::carl-1
aws:rds/subnetGroup:SubnetGroup                      urn:pulumi:dev::ersan-rds-passwords::aws:rds/subnetGroup:SubnetGroup::example
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:NatGateway$aws:ec2/natGateway:NatGateway::carl-0
aws:ec2/route:Route                                  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/route:Route::carl-private-1-nat-1
aws:ec2/route:Route                                  urn:pulumi:dev::ersan-rds-passwords::awsx:x:ec2:Vpc$awsx:x:ec2:Subnet$aws:ec2/route:Route::carl-private-0-nat-0
aws:rds/cluster:Cluster                              urn:pulumi:dev::ersan-rds-passwords::aws:rds/cluster:Cluster::carl-cluster-demo
aws:rds/clusterInstance:ClusterInstance              urn:pulumi:dev::ersan-rds-passwords::aws:rds/clusterInstance:ClusterInstance::carlInstances-1
aws:rds/clusterInstance:ClusterInstance              urn:pulumi:dev::ersan-rds-passwords::aws:rds/clusterInstance:ClusterInstance::carlInstances-0


Found no pending operations associated with dev

Backend
Name           pulumi.com
URL            https://app.pulumi.com/caboog
User           caboog
Organizations  caboog, pulumi

Dependencies:
NAME            VERSION
@pulumi/pulumi  3.44.2
@pulumi/random  4.8.2
@types/node     14.18.32
@pulumi/aws     5.18.0
@pulumi/awsx    0.40.1

Pulumi locates its logs in /var/folders/94/6jwsj_gd3z570l42tt1p4tdc0000gn/T/ by default
warning: A new version of Pulumi is available. To upgrade from version '3.37.2' to '3.44.1', run
   $ brew upgrade pulumi
or visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.
```

### Additional context

Code to reproduce the issue: https://github.com/caboog/pulumi-examples/tree/main/typescript/aws/ersan-rds-passwords

Related customer issue: https://pulumi.zendesk.com/agent/tickets/1910

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by er-san** [comment]
thanks, let me know if there is something I can help with.

**Comment by roothorp** [comment]
Hi @caboog - thanks for the issue. I've added this to our backlog to be prioritized in upcoming iterations.

**Comment by mnlumi** [comment]
hey @caboog The sample code here is using some old versions of aws and awsx. Do you believe this issue is still outstanding? I've not been able to get far enough in a reproduction with the sample code to confirm either way.

**Comment by VenelinMartinov** [comment]
I've tested this with a modified version of the program, included below and the password updates correctly.

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as awsx from "@pulumi/awsx";
import * as random from "@pulumi/random";
import { output } from "@pulumi/pulumi";

const vpc = new awsx.ec2.Vpc("carl", {
    cidrBlock: "172.24.0.0/20",
    enableDnsHostnames: true,
    enableDnsSupport: true,
})

export const vpcAZ = vpc.subnets[0].availabilityZone
const password = new random.RandomPassword("password", {
    length: 18,
    special: true,
    overrideSpecial: `_%@/"'`,
});


const sg = new aws.rds.SubnetGroup("example", {
    subnetIds: vpc.publicSubnetIds,
});
const defaultSecGroup = new aws.ec2.SecurityGroup("default", {
    vpcId: vpc.vpcId,
    ingress: [{
        protocol: "-1",
        self: true,
        fromPort: 0,
        toPort: 0,
    }],
    egress: [{
        fromPort: 0,
        toPort: 0,
        protocol: "-1",
        cidrBlocks: ["0.0.0.0/0"],
    }],
});


const carl_cluster_demo = new aws.rds.Cluster("carl-cluster-demo", {
    applyImmediately: true,
    availabilityZones: [
        "us-west-2a",
        "us-west-2b",
        "us-west-2c",
    ],
    backupRetentionPeriod: 7,
    copyTagsToSnapshot: true,
    dbClusterParameterGroupName: "default.aurora-postgresql14",
    dbSubnetGroupName: sg.name,
    engine: "aurora-postgresql",
    engineVersion: "14.4",
    masterUsername: "carl",
    masterPassword: password.result,
    databaseName: "postgres",
    networkType: "IPV4",
    port: 5432,
    preferredBackupWindow: "05:15-05:45",
    preferredMaintenanceWindow: "sat:08:25-sat:08:55",
    skipFinalSnapshot: true,
    vpcSecurityGroupIds: [defaultSecGroup.id],
}, {
    protect: false,
    deleteBeforeReplace: true,
});
for (const range = { value: 0 }; range.value < 2; range.value++) {
    const readOnlyClusterInstance = new aws.rds.ClusterInstance(`carlInstances-${range.value}`, {
        clusterIdentifier: carl_cluster_demo.id,
        instanceClass: "db.t3.medium",
        engine: "aurora-postgresql",
        engineVersion: carl_cluster_demo.engineVersion,
        applyImmediately: true,
        publiclyAccessible: true,
    });
}
export const dbpass = password.result

export const dbEndpoint = carl_cluster_demo.endpoint
```

**Comment by er-san** [comment]
@VenelinMartinov could you perhaps post when this was fixed? what version of the dependencies you are using? was this an **update** to the password or the initial creation of the cluster with the password? Example: create cluster with password A, update cluster with password B. Initial creation always worked fine, the bug was that an update to the password for a cluster that already existed didnt actually update.

**Comment by VenelinMartinov** [comment]
@er-san I am using the following plugin verisons:
```
Dependencies:
NAME            VERSION
@pulumi/aws     6.13.3
@pulumi/awsx    2.3.0
@pulumi/pulumi  3.96.2
@pulumi/random  4.15.0
@types/node     18.19.3
```

The situation I tested was as you described:
1. Provision with the initial password.
2. Verify password works by connecting via `psql`
3. Update password.
4. Verify new password works by connecting via `psql`

Have I missed some detail in my test?
